src/services/utilities.ts

```diff
@@ -1332,19 +1332,20 @@ namespace ts {
             return n;
         }
 
-        const candidate = findRightmostChildNodeWithTokens(children, /*exclusiveStartPosition*/ children.length, sourceFile);
+        const candidate = findRightmostChildNodeWithTokens(n, /*exclusiveStartPosition*/ children.length, sourceFile);
         return candidate && findRightmostToken(candidate, sourceFile);
     }
 
     /**
-     * Finds the rightmost child to the left of `children[exclusiveStartPosition]` which is a non-all-whitespace token or has constituent tokens.
+     * Finds the rightmost child to the left of `n.children[exclusiveStartPosition]` which is a non-all-whitespace token or has constituent tokens.
      */
-    function findRightmostChildNodeWithTokens(children: Node[], exclusiveStartPosition: number, sourceFile: SourceFile): Node | undefined {
+    function findRightmostChildNodeWithTokens(n: Node, exclusiveStartPosition: number, sourceFile: SourceFile): Node | undefined {
+        const children = n.getChildren(sourceFile);
         for (let i = exclusiveStartPosition - 1; i >= 0; i--) {
             const child = children[i];
 
-            if (isWhiteSpaceOnlyJsxText(child)) {
-                Debug.assert(i > 0, "`JsxText` tokens should not be the first child of `JsxElement | JsxSelfClosingElement`");
+            if (i === 0 && isWhiteSpaceOnlyJsxText(child)) {
+                Debug.assert(!(isJsxElement(n) || isJsxSelfClosingElement(n)), "`JsxText` tokens should not be the first child of `JsxElement | JsxSelfClosingElement`");
             }
             else if (nodeHasTokens(children[i], sourceFile)) {

```

Comment:
Out of scope, but should `children[i]` be `child` here and below?

Reply:
Yeah, this seems like an oops. I can fix this, if desired. Shouldn't change anything.

Reply:
Maybe as a follow-up.